File: C++/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int sz = nums.size();
        vector<int> dp(sz);
        dp[0] = nums[0];
        for (int i = 1; i < sz; i++) {
            dp[i] = max(nums[i], nums[i] + dp[i-1]);
        }
        return *max_element(dp.begin(), dp.end());
    }
};

File: C#/maximum-subarray.cs
public class Solution {
    public int MaxSubArray(int[] nums) {
        int acc = 0, max = nums[0];
        acc = nums[0];
        for (int i = 1; i < nums.Length; i++) {
            acc = Math.Max(nums[i], nums[i] + acc);
            max = Math.Max(acc, max);
        }
        return max;
    }
}